packages/dds/merge-tree/src/mergeTree.ts

```diff
@@ -1217,6 +1264,12 @@ export class MergeTree {
             const deltaSegments: IMergeTreeSegmentDelta[] = [];
             pendingSegmentGroup.segments.map((pendingSegment) => {
                 const overlappingRemove = !pendingSegment.ack(pendingSegmentGroup, opArgs);
+                if (opArgs.op.type === MergeTreeDeltaType.INSERT && this.options?.trackAttribution) {
```

Comment:
would it make sense to put this logic inside pendingSegment.ack? i guess we don't have options there, but we could probably find some solution.

Reply:
i originally had it there, but moved it out for exactly that reason. Maybe a reasonable third option is to add an optional `attributionKey` argument to `ack`? that'd give better encapsulation.

Reply:
I think to support the merge-tree extensibility point we've discussed, we're going to want to take that approach. I think `ack` should take an optional third argument which is roughly some declarative information about any attribution information that came alongside the ack. that bit will be injectable by something that can be configured on merge-tree construction. I'm imagining apps will use that part to inject their own custom logic about attributing certain props differently or adding back-compat or something.

I don't really want to settle on the API we add in this PR though, so to avoid back-compat problems I'm going to revert to something like this.